### PR TITLE
Seperate logic

### PR DESCRIPTION
**Pull request by jeffrey-dot-li**

### CODE REVIEW

File: lib/src/commands/add/file.dart [hunk 1]
```diff
@@ -1,45 +1,27 @@
...
-        properties[k is String ? k.trim() : k.toString().trim()] =
-            v is String ? v.trim() : v.toString().trim());
-    var result = await this.databaseHandler.addEntry(properties);
+    final AddDataModel addDataModel = new AddDataModel();
+    this.results = await addDataModel.addFileData(argResults.rest[0]);
     print('Added new entry \n \n');
     if (argResults['verbose'] == true) {
-      print(properties);
-      print(result);
+      print(this.results.toString() + '\n \n');
```
Comment (seendsouza) [inline lib/src/commands/add/file.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/commands/add/file.dart [hunk 2]
```diff
@@ -1,45 +1,27 @@
...
+    } else {
+      print(this.results['data'].toString() + '\n \n');
```
Comment (seendsouza) [inline lib/src/commands/add/file.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/commands/find/data.dart [hunk 3]
```diff
@@ -1,25 +1,29 @@
...
   run() async {
-    Map filters = parseArgsJson(argResults.rest[0]);
-    var results = await this.databaseHandler.getMatches(filters);
-    print(filters);
-    print(results);
+    final FindDataModel findDataModel = new FindDataModel();
+    this.results = await findDataModel.findStringData(argResults.rest[0]);
+    print('Found entries: \n \n');
+    if (argResults['verbose'] == true) {
+      print(this.results.toString() + '\n \n');
```
Comment (seendsouza) [inline lib/src/commands/find/data.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/commands/find/data.dart [hunk 4]
```diff
@@ -1,25 +1,29 @@
...
+    } else {
+      print(this.results['data'].toString() + '\n \n');
```
Comment (seendsouza) [inline lib/src/commands/find/data.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/commands/find/file.dart [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    argParser..addFlag('verbose', defaultsTo: false);
+  }
+
+  @override
+  run() async {
+    final FindDataModel addDataModel = new FindDataModel();
+    this.results = await addDataModel.findFileData(argResults.rest[0]);
+    print('Entries found \n \n');
+    if (argResults['verbose'] == true) {
+      print(this.results.toString() + '\n \n');
```
Comment (seendsouza) [inline lib/src/commands/find/file.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/commands/find/file.dart [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    } else {
+      print(this.results['data'].toString() + '\n \n');
```
Comment (seendsouza) [inline lib/src/commands/find/file.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/database/database.dart [hunk 7]
```diff
@@ -32,16 +60,19 @@ class DatabaseHandler {
...
   Future getMatches(properties) async {
+    Map results = new Map.from(this.resultFields);
     List<Filter> filters = List();
     properties.forEach((k, v) => filters.add(Filter.matches(k, v)));
-    print(filters);
     var records =
         (await this.store.findRecords(Finder(filter: Filter.and(filters))));
-    return records;
+    results['data'] = records;
+    results['status'] = 200;
```
Comment (seendsouza) [inline lib/src/database/database.dart]: https://vscout.readthedocs.io/en/latest/contributing/style-guide.html#no-magic-numbers

File: lib/src/database/database.dart [hunk 8]
```diff
@@ -1,29 +1,57 @@
...
   }
 
   InitializeDb() async {
-    this.db = await this.newDb();
+    this.resultFields = new Map();
+    this.resultFields['status'] = null;
+    this.resultFields['data'] = null;
+    //Set the path to the database
+    this.relativeDatabasePath = '/../database/vscout.db';
+    this.realDatabasePath = (dirname(Platform.script.toFilePath()).toString() +
```
Comment (seendsouza) [inline lib/src/database/database.dart]: absoluteDatabasePath

File: lib/src/database/database.dart [hunk 9]
```diff
@@ -1,29 +1,57 @@
...
 
-String dbPath = '/../database/vscout.db';
-var pathToDb = (dirname(Platform.script.toFilePath()).toString() + dbPath);
-
 class DatabaseHandler {
   Database db;
   Store store;
+  File databaseFile;
+  String relativeDatabasePath;
+  String realDatabasePath;
```
Comment (seendsouza) [inline lib/src/database/database.dart]: absoluteDatabasePath

File: lib/src/database/database.dart [hunk 10]
```diff
@@ -50,6 +81,8 @@ class DatabaseHandler {
     String key = uuid.v4();
     Record record = Record(store, entry, key);
     record = await this.db.putRecord(record);
-    return '${record.toString()}';
+    results['data'] = record;
+    results['status'] = 200;
```
Comment (seendsouza) [inline lib/src/database/database.dart]: https://vscout.readthedocs.io/en/latest/contributing/style-guide.html#no-magic-numbers

File: lib/src/database/database.dart [hunk 11]
```diff
@@ -1,29 +1,57 @@
...
+  }
+
+  DatabaseHandler._internal() {
+    //
   }
 
   InitializeDb() async {
-    this.db = await this.newDb();
+    this.resultFields = new Map();
+    this.resultFields['status'] = null;
```
Comment (seendsouza) [inline lib/src/database/database.dart]: https://dart.dev/guides/language/effective-dart/usage#dont-use-an-explicit-default-value-of-null

File: lib/src/models/add/data.dart [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    this.result = await this.databaseHandler.addEntry(dataEntry);
+    this.result['query'] = dataEntry.toString();
+    this.result['queryType'] = 'ADD/DATA/MAP';
+    return this.result;
+  }
+
+  addFileData(String relativeFilePath) async {
+    String fileFolder = '/../files/';
+    //  Tries to find file in [files] folder.
+    var absFilePath = (dirname(Platform.script.toFilePath()).toString() +
```
Comment (seendsouza) [inline lib/src/models/add/data.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

File: lib/src/models/find/data.dart [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:path/path.dart';
+
+import '../../utils/utils.dart';
+import '../model.dart';
+
+class FindDataModel extends Model {
+  findStringData(String dataEntry) async {
```
Comment (seendsouza) [inline lib/src/models/find/data.dart:10]: I've seen similar code in other models, so should we move this code to the [Model] class?

File: lib/src/models/find/data.dart [hunk 14]
```diff
@@ -0,0 +1,46 @@
...
+    this.result = await await this.databaseHandler.getMatches(dataEntry);
+    this.result['query'] = dataEntry.toString();
+    this.result['queryType'] = 'FIND/DATA/MAP';
+    return this.result;
+  }
+
+  findFileData(String relativeFilePath) async {
+    String fileFolder = '/../files/';
+    //  Tries to find file in [files] folder.
+    var absFilePath = (dirname(Platform.script.toFilePath()).toString() +
```
Comment (seendsouza) [inline lib/src/models/find/data.dart]: https://dart.dev/guides/language/effective-dart/usage#strings

**Review by seendsouza — CHANGES_REQUESTED** [review]

File: lib/src/models/find/data.dart (hunk 13 above)
Comment (jeffrey-dot-li) [reply lib/src/models/find/data.dart:10]: That can be done in a future pr?

File: lib/src/commands/add/data.dart [hunk 15]
```diff
@@ -20,9 +20,9 @@ class DataCommand extends Command {
     this.results = await addDataModel.addStringData(argResults.rest[0]);
     print('Added new entry \n \n');
     if (argResults['verbose'] == true) {
-      print(this.results.toString() + '\n \n');
+      print('${this.results.toString()} \n \n');
     } else {
-      print(this.results['data'].toString() + '\n \n');
+      print("${this.results['data'].toString()}+ \n \n");
```
Comment (seendsouza) [inline lib/src/commands/add/data.dart:25]: wait why is there a +

File: lib/src/commands/add/file.dart [hunk 16]
```diff
@@ -19,9 +19,9 @@ class FileCommand extends Command {
     this.results = await addDataModel.addFileData(argResults.rest[0]);
     print('Added new entry \n \n');
     if (argResults['verbose'] == true) {
-      print(this.results.toString() + '\n \n');
+      print('${this.results.toString()} \n \n');
     } else {
-      print(this.results['data'].toString() + '\n \n');
+      print("${this.results['data'].toString()}+ \n \n");
```
Comment (seendsouza) [inline lib/src/commands/add/file.dart:24]: again

File: lib/src/commands/find/data.dart [hunk 17]
```diff
@@ -20,9 +20,9 @@ class DataCommand extends Command {
     this.results = await findDataModel.findStringData(argResults.rest[0]);
     print('Found entries: \n \n');
     if (argResults['verbose'] == true) {
-      print(this.results.toString() + '\n \n');
+      print('${this.results.toString()} \n \n');
     } else {
-      print(this.results['data'].toString() + '\n \n');
+      print("${this.results['data'].toString()}+ \n \n");
```
Comment (seendsouza) [inline lib/src/commands/find/data.dart:25]: rip

File: lib/src/commands/find/file.dart [hunk 18]
```diff
@@ -19,9 +19,9 @@ class FileCommand extends Command {
     this.results = await addDataModel.findFileData(argResults.rest[0]);
     print('Entries found \n \n');
     if (argResults['verbose'] == true) {
-      print(this.results.toString() + '\n \n');
+      print('${this.results.toString()} \n \n');
     } else {
-      print(this.results['data'].toString() + '\n \n');
+      print("${this.results['data'].toString()}+ \n \n");
```
Comment (seendsouza) [inline lib/src/commands/find/file.dart:24]: F, but this code seems to be reused a lot. u think a command class would be good (another pr)

Comment (seendsouza) [reply lib/src/commands/find/file.dart:24]: also fix conflicts

File: lib/src/models/find/data.dart (hunk 13 above)
Comment (seendsouza) [reply lib/src/models/find/data.dart:10]: yeah

**Review by seendsouza — APPROVED** [review]

**Pull request merged**